packages/http-server/test/integration/http-server.integration.ts

```diff
@@ -143,9 +175,45 @@ describe('HttpServer (integration)', () => {
     await server.stop();
   }
 
-  function getAsync(url: string): Promise<IncomingMessage> {
+  function getAsync(urlString: string): Promise<IncomingMessage> {
+    return new Promise((resolve, reject) => {
+      get(urlString, resolve).on('error', reject);
+    });
+  }
+
+  function givenHttpsServer({
+    usePfx,
+    host,
+  }: {
+    usePfx?: boolean;
+    host?: string;
+  }): HttpServer {
+    const options: HttpServerOptions = {protocol: 'https', host};
+    if (usePfx) {
+      const pfxPath = path.join(__dirname, 'pfx.pfx');
+      options.pfx = fs.readFileSync(pfxPath);
+      options.passphrase = 'loopback4';
+    } else {
+      const keyPath = path.join(__dirname, 'key.pem');
+      const certPath = path.join(__dirname, 'cert.pem');
+      options.key = fs.readFileSync(keyPath);
+      options.cert = fs.readFileSync(certPath);
+    }
+    return new HttpServer(dummyRequestHandler, options);
+  }
+
+  function httpsGetAsync(urlString: string): Promise<IncomingMessage> {
+    const agentOptions = {
+      agent: new https.Agent({
+        rejectUnauthorized: false,
+      }),
+    };
+
+    const urlOptions = url.parse(urlString);
+    const options = {...agentOptions, ...urlOptions};
```

Comment:
A possibly subjective nitpick - feel free to ignore.

I think the following version is more concise and possibly easier to understand.

```ts
const agent = new https.Agent(...);
const urlOptions = //...
const options = {agent, ...urlOptions};
```